prepack.mjs

```diff
@@ -41,3 +45,46 @@ ${Object.keys(model)
 
   return dFile
 }
+// Generate types for the Categories and Subcategories dottedNames
+const categories = [
+  'transport',
+  'alimentation',
+  'logement',
+  'divers',
+  'services sociétaux'
+]
+function generateCategoriesTypes(destPath) {
+  const rawData = fs.readFileSync(destPath)
+
+  const dFile = `
+export type Categories = ${categories.map((category) => `  | "${category}"`).join('\n')}
+`
+
+  return dFile
+}
+
+function generateSubcategoriesTypes(destPath) {
+  const rawData = fs.readFileSync(destPath)
+  const model = JSON.parse(rawData)
+
+  const subcategories = categories.flatMap((category) =>
+    Object.keys(model).filter(
+      (dottedName) =>
+        dottedName.startsWith(category + ' . ') &&
+        dottedName.split(' . ').length === 2
+    )
+  )
+
+  const dFile = `
+export type Subcategories =
+${subcategories.map((subcategory) => `  | "${subcategory}"`).join('\n')}
+`
+
+  return dFile
+}
+
+fs.writeFileSync('./types/categories.d.ts', generateCategoriesTypes(destPath))
+fs.writeFileSync(
+  './types/subcategories.d.ts',
+  generateSubcategoriesTypes(destPath)
+)
```

Comment:
Sync, sync, sync everywhere ! We should have a main async function and use sf promises

Reply:
Salut @paulsouche, j'ai pas mal contribué sur cette base de code et en particulier sur les scripts de compilation du modèle.

Je ne suis pas très calé en JS et me demandais pourquoi il serait préférable d'utiliser des _promises_ ici ? Le code étant séquentiel et les fichiers petits, je ne vois pas trop ce que ça apporterait :thinking:

Reply:
Hello ! Tu peux faire plusieurs opérations en parallèle au lieu de bloquer le thread